src/js/controller/controller.js

```diff
@@ -1310,6 +1310,16 @@ Object.assign(Controller.prototype, {
         } catch (error) {
             return Promise.reject(error);
         }
+
+        // Under very special conditions, we enable the html5 looping attribute
+        if (playlist.length === 1
```

Comment:
Is it possible to do this in the HTML5 provider?

Reply:
I was trying to keep the provider out of worrying about the playlist level info - to keep responsibilities separate. I know the model is available pretty much everywhere but it feels wrong. I'll try it and see what it's like.

Reply:
The controller is pretty high level for something specific to mp4 sources, especially given the additional constraints (it's essentially for API-level functionality). HTML5 provider's setPlaylistItem function might be a good place to put it. When you say the using the model feels wrong, what do you mean?